utils/zerovec/derive/src/varule.rs

```diff
@@ -86,6 +86,19 @@ pub fn derive_impl(
         quote!(<#unsized_field as zerovec::ule::VarULE>::validate_byte_slice(last_field_bytes)?;)
     };
 
+    let attr_path: Path = syn::parse_str("varule").unwrap();
+    let validate_with = match utils::find_validate_with_path(&input.attrs, &attr_path) {
+        Ok(Some(fn_path)) => {
+            quote! {
+                #fn_path(unsafe { Self::from_byte_slice_unchecked(bytes) })?;
```

Comment:
thought: for ULE this should be free but for VarULE this will be a couple extra instructions. i suppose we don't care about that?

this way of doing it does make the derive code simpler.

Reply:
What are you comparing against?

Note that the case where `validate_with` is not used, there is no code being added.

Reply:
oh, sorry, here we construct an `&Self` (involving a minor amount of ptr metadata arithmetic) and immediately throw it away, we could _theoretically_ merge it with the `last_field_validator` but doing that consistently and well will be a mess so it doesn't really matter anyway